### PR TITLE
`<spc b Y>` doesn't change cursor position

### LINKED ISSUE #117: Vim remap suggestion for `b.Y`
## Problem

Currently, I don't like that `SPC b Y` moves my cursor away from where I am currently positioned. That mapping is currently:

```
{
	"key": "Y",
	"name": "Copy buffer to clipboard",
	"type": "commands",
	"commands": [
		"editor.action.selectAll",
		"editor.action.clipboardCopyAction",
		"cancelSelection"
	]
}
```

## Solution

To fix this, I made the following remapping:

```
{
  "command": "vim.remap",
  "keys": "b.Y",
  "name": "Copy buffer to clipboard",
  "type": "command",
  "args": {
    "commands": [
      {
        "command": ":%y"
      }
    ]
  }
}
```

Works as you'd expect (you'll see the number of lines yanked in the status bar). Wanted to get an opinion on this before I submitted it as a PR, since it feels a bit more elegant.

### PR DESCRIPTION
**Pull request by MarcoIeni**
Close #117

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Fix the issue where config default setting command normalized existing vim's keybindings (e.g. `<space>` to ` `) and can cause duplicate keybindings.
+- `<spc b Y>` doesn't change cursor position
```
Comment (stevenguh) [inline CHANGELOG.md:21]: nit: `<spc> b Y`
Maybe it's clearer to say "Keep cursor position after `<spc> b Y`"?

Comment (MarcoIeni) [reply CHANGELOG.md:21]: Yes, I will update this

**Comment by macintacos** [comment]
Hey folks, hate to get in the middle of this, but this isn't "perfect". I ran into an issue with this mapping when I had a file open, opened a new file, and then invoked `SPC b Y`. With this mapping, `<C-o>` brings me back to the previous file instead of leaving me in the file I just opened. It felt a bit jarring. Not sure how often folks would run into this but... perhaps there's something else that can be done?

**Comment by macintacos** [comment]
Quick and dirty alternative is to make a `mark` at the current position, do the whole `ggVGy`, and then return to the mark that was made. Maybe something like this for the keybinding?

```
mxggVGy'x
```

**Comment by stevenguh** [comment]
Thank you for pointing that out. For some reason, I can't reproduce it. Anyway, I think it is important that `<spc> b Y` doesn't return cursor to unexpected places. It would be a worse experience than not returning imo. Using mark has to con of wiping whatever on that mark. Hmmmm...

**Comment by MarcoIeni** [comment]
> For some reason, I can't reproduce it.

I was able to reproduce it.

Steps:
- open a file
- press something like `5j` to move away from the top
- press `gg` to go to the top
- press `<spc> b Y`

Now you are on line 5, while expected behavior is to stay at the top.

> sing mark has to con of wiping whatever on that mark.

I agree that setting a mark may be too invasive.

This is a rare corner case that doesn't happen so much I think and going back to the top is really easy.
At the moment I can't think of a better solution than this one.

**Comment by macintacos** [comment]
Again, this is a tad hacky, but you could make the mark a character that is very unlikely to be used by the user, such as something like `≈` (which for me is "option+x" on the Mac). I just tested that and at least _that_ works. But again, a couple drawbacks:

- I don't have a Linux/Windows setup, so I'm not sure if those characters would be supported on those platforms
- It's not as elegant as something like `:%y` (which we can't use because you'd need to turn on neovim integration)
- There is still a (smaller IMO) chance that a user could be using a character like `≈` for marks and get unexpected behavior

**Comment by MarcoIeni** [comment]
I understand your point. 
One additional con of your approach is that when the user will list all marks it will be confused by this additional mark.

I still prefer the buggy approach because it doesn't mess up with user stuff. Steven, what do you vote for? :)

**Comment by stevenguh** [comment]
Another option is to implement our custom command like https://github.com/VSpaceCode/VSpaceCode/blob/master/src/pathCommands.ts

Custom command will allow us to copy the whole buffer to clipboard pretty easily while maintaining the cursor position. However, it won't yank to any register in vim like the current command, and as well as not showing any animation of yank in vim with `vim.highlightedyank.enable` on like the current command as well.

**Comment by MarcoIeni** [comment]
ok, so I merge this?

**Comment by stevenguh** [comment]
Sorry was not being clear. I am proposing to use a custom command solve the cursor issue since that shouldn't have any weird edge cases. I can try it tomorrow and see which one works the way we like?

**Comment by MarcoIeni** [comment]
Oh, ok. Yes, sure!

**Comment by stevenguh** [comment]
Just pushed a commit that uses a custom implementation. That works well for the purpose of copying the buffer to the clipboard without changing the cursor position

**Comment by MarcoIeni** [comment]
For me it works fine, but macintacos is the real bug hunter here :rofl:

**Comment by macintacos** [comment]
Heh, I think this works well too. If I run into any weird edge cases I'll just open another issue 🙂

**Comment by stevenguh** [comment]
Let's try this then :)

**Pull request merged**